internal/provider/resource_ger_ruleset_rules_ordering.go

```diff
@@ -0,0 +1,189 @@
+	d.Set("alert_source_shortname", alertSourceShortName)
+	d.Set("alert_source_version", alertSourceVersion)
+	d.Set("ger_id", gerID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceGERRulesetRulesOrderingRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*api.Client)
+
+	tflog.Info(ctx, "Reading GER Ruleset Rule", tf.M{
+		"id": d.Id(),
+	})
+
+	alertSourceData := api.GERAlertSource{
+		Name:    d.Get("alert_source_shortname").(string),
+		Version: d.Get("alert_source_version").(string),
+	}
+
+	gerRulesetRules, err := client.GetGERRulesetByAlertSource(ctx, d.Get("ger_id").(string), alertSourceData)
+	if err != nil {
+		if api.IsResourceNotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	gerRulesetRulesOrdering := &api.GERReorderRulesetRules{
+		ID:       gerRulesetRules.ID,
+		GER_ID:   gerRulesetRules.GER_ID,
+		Ordering: gerRulesetRules.Ordering,
+	}
+	if err = tf.EncodeAndSet(gerRulesetRulesOrdering, d); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceGERRulesetRulesOrderingUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*api.Client)
+
+	req := &api.GERReorderRulesetRulesReq{}
+
+	alertSource := d.Get("alert_source").(string)
+	alertSources, err := client.ListAlertSources(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	alertSourceShortName, alertSourceVersion := "", ""
+	isValidAlertSource := false
+	for _, alertSourceData := range alertSources {
+		if alertSourceData.Type == alertSource {
+			alertSourceShortName = alertSourceData.ShortName
+			alertSourceVersion = alertSourceData.Version
+			isValidAlertSource = true
+			break
+		}
+	}
+	if !isValidAlertSource {
+		return diag.Errorf("%s is not a valid alert source name. Find all alert sources on https://www.squadcast.com/integrations", alertSource)
+	}
+
+	ordering := d.Get("ordering").([]interface{})
+
+	orderingList := make([]uint, len(ordering))
+	for i, v := range ordering {
+		val, err := strconv.ParseUint(v.(string), 10, 64)
+		if err != nil {
+			return diag.Errorf("Invalid rule id.")
+		}
+		orderingList[i] = uint(val)
+	}
+	req.Ordering = orderingList
+
+	tflog.Info(ctx, "Updating GER Ruleset Rule Ordering", tf.M{
+		"req": req,
+	})
+
+	gerRulesetRulesOrdering, err := client.UpdateGERRulesetRulesOrdering(ctx, d.Get("ger_id").(string), api.GERAlertSource{
+		Name:    alertSourceShortName,
+		Version: alertSourceVersion,
+	}, req)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := strconv.FormatUint(uint64(gerRulesetRulesOrdering.ID), 10)
+	d.SetId(id)
+	d.Set("alert_source_shortname", alertSourceShortName)
+	d.Set("alert_source_version", alertSourceVersion)
+
+	return resourceGERRulesetRulesOrderingRead(ctx, d, meta)
+}
+
+// set state to null
+func resourceGERRulesetRulesOrderingDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	return nil
```

Comment:
Shouldn't we set the state to nil as well?

Reply:
This handles that